frontend/src/components/Chat/ChatsGroup.js

```diff
@@ -33,39 +22,95 @@ const Conversation = ({ participants, onClick, selected, lastMessage }) => {
+      </ListItem>
   );
 };
 
-const ChatsGroup = ({ userData }) => {
+const pollingState = {
+  pollingActive: false,
+  pollingRunning: false,
+  selDialogMessages: [],
+  rooms: [],
+  roomId: null
+};
+
+const ChatsGroup = ({userData}) => {
   const [allRooms, setAllRooms] = useState([]);
   const [selectedDialog, setSelDialog] = useState([]);
-  const [selRoomId, setSelRoomId] = useState({ id: null, participants: null });
+  const [selRoomId, setSelRoomId] = useState({id: null, participants: null});
 
   useEffect(() => {
-    getAllRooms().then((response) => setAllRooms(response.content));
+    getAllRooms().then((response) => {
+      setAllRooms(response.content);
+      pollingState.rooms = response.content;
+      pollingState.pollingActive = true;
+      if (!pollingState.pollingRunning) {
+        pollNext();
+      }
+    });
+
+    return function stopPolling() {
+      pollingState.pollingActive = false;
+    };
   }, []);
 
-  const clickHandler = (id, participants) => {    
-    setSelRoomId({ id, participants });
-    
-    getRoom(id).then((response) => setSelDialog(response.content.reverse()));
+  const pollNext = () => {
+    if (!pollingState.pollingActive) return;
+    pollingState.pollingRunning = true;
+    poll().then(rooms => {
+      pollingState.pollingRunning = false;
+      if (!pollingState.pollingActive) return;
+
+      const newRoomState = [];
+      rooms.forEach(function (room) {
+        // update current room messages if its opened
+        if (room.id === pollingState.roomId) {
+          pollingState.selDialogMessages = [...pollingState.selDialogMessages, ...room.messages];
+          setSelDialog([...pollingState.selDialogMessages]); // insert new message
+        }
+
+        newRoomState.push(room);
+      });
+
+      // copy old rooms to the new state excluding just updated
+      pollingState.rooms.filter(room => !newRoomState.some(item => item.id === room.id))
+          .forEach(room => newRoomState.push(room));
+
+      setAllRooms(newRoomState);
+      pollingState.rooms = newRoomState;
+
+      // here possible to show notification
+      // todo: refactor this to dispatched events, possible call stack exceeded exception?
+      pollNext();
+    });
+  };
+
+  const clickHandler = (id, participants) => {
+    setSelRoomId({id, participants});
+    pollingState.roomId = id;
+
+    getRoom(id).then((response) => {
+      const messages = response.content.reverse();
+      pollingState.selDialogMessages = messages;
+      setSelDialog(messages);
+    });
   };
 
   const sendMessage = (message) => {
     sendMessageRoom({
       room: selRoomId.id,
-      type: "TEXT",
-      source: "USER",
-      body: message,
-    }).then((response) => setSelDialog([...selectedDialog, response]));
+      type: 'TEXT',
+      source: 'USER',
+      body: message
+    }).then((response) => console.log('Message sent:' + response.body) /*setSelDialog([...selectedDialog, response])*/);
```

Comment:
Is `setSelDialog` should be commented?

Reply:
we shouldn't add message from the response here, as it will be received in the polling call, here we can just handle errors if its occurs. but for now I not sure what we can show there. I'll remove this comment for now.